recipes/libcoap/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibCoapConan(ConanFile):
+    name = "libcoap"
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/gocarlos/libcoap/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """A CoAP (RFC 7252) implementation in C"""
+    topics = ("coap")
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_epoll": [True, False],
+        "tls_backend": [None, "with_openssl", "with_gnutls", "with_tinydtls", "with_mbedtls"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_epoll": False,
+        "tls_backend": "with_openssl",
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        if self.options.tls_backend == "with_openssl":
+            self.requires.add("openssl/1.1.1d")
+        if self.options.tls_backend == "with_mbedtls":
+            self.requires.add("mbedtls/2.16.3-apache")
+        if self.options.tls_backend == "with_gnutls":
+            raise ConanInvalidConfiguration("gnu tls not available yet")
+        if self.options.tls_backend == "with_tinydtls":
+            raise ConanInvalidConfiguration("tinydtls not available yet")
+
+    def _patch_files(self):
+        if self.options.tls_backend == "with_openssl":
+            replace_ssl = 'OpenSSL::SSL'
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), replace_ssl, "OpenSSL::OpenSSL")
+            replace_crypto = 'OpenSSL::Crypto'
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), replace_crypto, "OpenSSL::OpenSSL")
```

Comment:
evlt. related to https://github.com/conan-io/conan/issues/5090 https://github.com/conan-io/conan/issues/2387

Reply:
Almost there: https://github.com/conan-io/conan-center-index/pull/1782

Reply:
@gocarlos it should be available soon, do you want to wait for #1782 or do you need this for now?

Reply:
lets wait for it

Reply:
was merged

Reply:
bump?

Reply:
done